compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3021,6 +3021,12 @@ public void visit(BLangErrorVarRef varRefExpr, AnalyzerData data) {
             refItem.isLValue = true;
             checkExpr(refItem, data);
 
+            if (types.isFunctionVarRef(refItem)) {
+                dlog.error(refItem.pos, DiagnosticErrorCode.INVALID_ASSIGNMENT_DECLARATION_FINAL,
+                        Names.FUNCTION);
+                unresolvedReference = true;
```

Comment:
This test (https://github.com/ballerina-platform/ballerina-lang/pull/38430/commits/f5d3e1a895e63072bfb813cdc3a32862588fd60f#diff-fe936e5cf31d19393d76ddb6a810f5b5fdd587ebd7d59e8523f71c76dfd2d814R39) covers this